src/libcore/ops/generator.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// The result of a generator resumption.
+///
+/// This enum is returned from the `Generator::resume` method and indicates the
+/// possible return values of a generator. Currently this corresponds to either
+/// a suspension point (`Yielded`) or a termination point (`Complete`).
+#[derive(Debug)]
+#[cfg_attr(not(stage0), lang = "generator_state")]
+#[unstable(feature = "generator_trait", issue = "43122")]
+pub enum State<Y, R> {
```

Comment:
Should this have a name other than `State` if it's going to be exported as `std::ops::State`?